src/python/grpcio_test/grpc_test/_adapter/_low_test.py

```diff
@@ -211,5 +211,76 @@ def testEcho(self):
     del server_call
 
 
+class HangingServerShutdown(unittest.TestCase):
+
+  def setUp(self):
+    self.server_completion_queue = _low.CompletionQueue()
+    self.server = _low.Server(self.server_completion_queue, [])
+    self.port = self.server.add_http2_port('[::]:0')
+    self.client_completion_queue = _low.CompletionQueue()
+    self.client_channel = _low.Channel('localhost:%d'%self.port, [])
+
+    self.server.start()
+
+  def tearDown(self):
```

Comment:
This tearDown method doesn't match its corresponding setUp - a setUp/tearDown pair should always be able to function in the presence of someone inserting an empty testUpDown method into the test, but in this case self.server_completion_queue is shut down during the test method.

But hey, it's idempotent, so I think that that means that you could if you wanted to shut it down in tearDown as well as in the test?